drift-protocol/src/main/java/com/facebook/drift/protocol/TBinaryProtocol.java

```diff
@@ -391,6 +391,22 @@ public ByteBuffer readBinary()
         return ByteBuffer.wrap(buf);
     }
 
+    @Override
+    public void readBinary(byte[] buf, int offset, int length)
+            throws TException
+    {
+        int size = checkSize(readI32());
+        if (size == length) {
+            while (length > 0) {
```

Comment:
Could you please extract it into an utility method? Maybe something like `readAllInBatches` ... TO avoid copy paste